### ISSUE TITLE
Support on Mac for cp2105

### ISSUE THREAD

**Issue by mimmoLinino** [nekromant/pl2303gpio#3]
Do you think this driver can work also on Mac? What are the steps to build it and install on Mac?
Do you think this driver can support as well different chip like CP2105 ?
Thanks

**Comment by nekromant** [comment]
1. Folks reported that it works fine on mac, just skip the udev rules part - mac has no udev
2. CP2105 may have different control transfers for GPIO, I have no hardware right now to test it - patches welcome.

**Comment by alfran** [comment]
I built the code on Linux Ubuntu and it works. Then I built the same code in MAC OS and is not working as it should.

I installed lgcc (Xcode native), libusub-1.0 and pkg-config with MacPorts instead.

The cp2103 and 2105 controllers have the same address for GPIO (the only difference is that you have to chose to connect to standard or extended port).

**Comment by nekromant** [comment]
@alfran Okay, looks like it's time to dust off my old hackintosh and see for myself what's going on. Meanwhile if you have docs on what cotrol transfers cp2105 uses - please post, I'll see if I can add them

**Comment by alfran** [comment]
@nekromant I need to use CP2105 extended port GPIO 0. The registers are the same for both ports (Extended and Standard) and are the same for CP2103 and CP2105. I'm ready to run some test for you if you need!  let me know if you need extra information.

**Comment by zoobab** [comment]
What about CP2104? I discovered that your tool was also supporting CPx devices, although the name is pl2303gpio. Maybe time to rename it?

I have a cp2104 gpio board at home, will try it out.

I am also condisering ordering some CP2112 boards (8 gpios), do you have any experience with those?

**Comment by nekromant** [comment]
Guys, I only have a few cp2103 devices around. CP2104 were on my TODO list somewhere, but the parcel from aliexpress  got lost somewhere on the way. Patches welcome.

**Comment by zoobab** [comment]
BTW, I just updated my PL2303 gpio sysfs page yesterday:

http://www.zoobab.com/pl2303hxd-gpio

If you could add a link...

**Comment by zoobab** [comment]
I also have this board at home based on PL2303TB:

http://www.oneping.com.tw/p_PL2303TB_PWM_GPIO.htm

It has 12 GPIOs to play with, I should give it a shot.

I also have the standalone chips to solder on a PCB, but need to find the time to solder them.

**Comment by nekromant** [comment]
> What about CP2104? I discovered that your tool was also supporting CPx devices, although the name is pl2303gpio. Maybe time to rename it?

@zoobab Yes, it was initially for pl2303hxa, but since I never managed to get external eeprom working for my app I had to switch to cp2103 and never took time to rename. 

As of your sysfs approach - it's way cleaner, but I needed this to run on a few devices where I have no kernel sources (mediatek) and will unlikely ever have. I hope yours will get eventually merged into mainline.

**Comment by dony71** [comment]
does it mean this tools working also for CP2104 without any modification?

**Comment by nekromant** [comment]
Folks, just letting you know, that I'm discontinuing these tool. I've made a replacement here: https://github.com/nekromant/pygpiotools

It's still in very early stage, but has windows support apart from linux. PL2303HXA is working on both platforms, cp210x is yet to come, but it will be soon.
